Okay, the failed jenkins build were for the previous update. 

MythSession.cpp: In constructor 'XFILE::CMythSession::CMythSession(const CURL&)':
MythSession.cpp:361:69: error: expected '{' before ':' token
MythSession.cpp: At global scope:
MythSession.cpp:361:69: error: expected unqualified-id before ':' token
make[1]: *** [MythSession.o] Error 1
make: *** [xbmc/filesystem/filesystem.a] Error 2

I think I fixed that already.

--- a/xbmc/filesystem/DllLibCurl.cpp
+++ b/xbmc/filesystem/DllLibCurl.cpp
@@ -165,7 +165,7 @@ void DllLibCurlGlobal::CheckIdle()
       it = m_sessions.erase(it);
       continue;
     }
-    it++;
+    ++it;
   }
 
   /* check if we should unload the dll */
@@ -182,7 +182,7 @@ void DllLibCurlGlobal::easy_aquire(const char *protocol, const char *hostname, C
   CSingleLock lock(m_critSection);
 
   VEC_CURLSESSIONS::iterator it;
-  for(it = m_sessions.begin(); it != m_sessions.end(); it++)
+  for(it = m_sessions.begin(); it != m_sessions.end(); ++it)
   {
     if( !it->m_busy )
     {
@@ -261,7 +261,7 @@ void DllLibCurlGlobal::easy_release(CURL_HANDLE** easy_handle, CURLM** multi_han
   }
 
   VEC_CURLSESSIONS::iterator it;
-  for(it = m_sessions.begin(); it != m_sessions.end(); it++)
+  for(it = m_sessions.begin(); it != m_sessions.end(); ++it)
   {
     if( it->m_easy == easy && (multi == NULL || it->m_multi == multi) )
     {
@@ -280,7 +280,7 @@ CURL_HANDLE* DllLibCurlGlobal::easy_duphandle(CURL_HANDLE* easy_handle)
   CSingleLock lock(m_critSection);
 
   VEC_CURLSESSIONS::iterator it;
-  for(it = m_sessions.begin(); it != m_sessions.end(); it++)
+  for(it = m_sessions.begin(); it != m_sessions.end(); ++it)
   {
     if( it->m_easy == easy_handle )
     {
@@ -305,7 +305,7 @@ void DllLibCurlGlobal::easy_duplicate(CURL_HANDLE* easy, CURLM* multi, CURL_HAND
     *multi_out = DllLibCurl::multi_init();
 
   VEC_CURLSESSIONS::iterator it;
-  for(it = m_sessions.begin(); it != m_sessions.end(); it++)
+  for(it = m_sessions.begin(); it != m_sessions.end(); ++it)
   {
     if( it->m_easy == easy )
     {

--- a/xbmc/filesystem/FTPParse.cpp
+++ b/xbmc/filesystem/FTPParse.cpp
@@ -38,7 +38,6 @@ using namespace std;
 
 CFTPParse::CFTPParse()
 {
-  m_name = "";
   m_flagtrycwd = 0;
   m_flagtryretr = 0;
   m_size = 0;

--- a/xbmc/filesystem/HTSPDirectory.cpp
+++ b/xbmc/filesystem/HTSPDirectory.cpp
@@ -91,7 +91,7 @@ CHTSPDirectorySession* CHTSPDirectorySession::Acquire(const CURL& url)
 {
   CSingleLock lock(g_section);
 
-  for(SSessions::iterator it = g_sessions.begin(); it != g_sessions.end(); it++)
+  for(SSessions::iterator it = g_sessions.begin(); it != g_sessions.end(); ++it)
   {
     if(it->hostname == url.GetHostName()
     && it->port     == url.GetPort()
@@ -129,7 +129,7 @@ void CHTSPDirectorySession::Release(CHTSPDirectorySession* &session)
     return;
 
   CSingleLock lock(g_section);
-  for(SSessions::iterator it = g_sessions.begin(); it != g_sessions.end(); it++)
+  for(SSessions::iterator it = g_sessions.begin(); it != g_sessions.end(); ++it)
   {
     if(it->session == session)
     {
@@ -156,7 +156,7 @@ void CHTSPDirectorySession::CheckIdle(DWORD idle)
       it = g_sessions.erase(it);
     }
     else
-      it++;
+      ++it;
   }
 }
 
@@ -329,7 +329,7 @@ SChannels CHTSPDirectorySession::GetChannels(STag& tag)
   SChannels channels;
 
   std::vector<int>::iterator it;
-  for(it = tag.channels.begin(); it != tag.channels.end(); it++)
+  for(it = tag.channels.begin(); it != tag.channels.end(); ++it)
   {
     SChannels::iterator it2 = m_channels.find(*it);
     if(it2 == m_channels.end())
@@ -375,7 +375,7 @@ bool CHTSPDirectory::GetChannels( const CURL &base
 
   SEvent event;
 
-  for(SChannels::iterator it = channels.begin(); it != channels.end(); it++)
+  for(SChannels::iterator it = channels.begin(); it != channels.end(); ++it)
   {
     if(!m_session->GetEvent(event, it->second.event))
       event.Clear();
@@ -439,7 +439,7 @@ bool CHTSPDirectory::GetDirectory(const CURL& url, CFileItemList &items)
 
     STags tags = m_session->GetTags();
     std::string filename, label;
-    for(STags::iterator it = tags.begin(); it != tags.end(); it++)
+    for(STags::iterator it = tags.begin(); it != tags.end(); ++it)
     {
       filename = StringUtils::Format("tags/%d/", it->second.id);
       label = StringUtils::Format("Tag: %s", it->second.name.c_str());

--- a/xbmc/filesystem/MythSession.cpp
+++ b/xbmc/filesystem/MythSession.cpp
@@ -67,7 +67,7 @@ void CMythSession::CheckIdle()
     }
     else
     {
-      it++;
+      ++it;
     }
   }
 }
@@ -77,7 +77,7 @@ CMythSession* CMythSession::AquireSession(const CURL& url)
   CSingleLock lock(m_section_session);
 
   vector<CMythSession*>::iterator it;
-  for (it = m_sessions.begin(); it != m_sessions.end(); it++)
+  for (it = m_sessions.begin(); it != m_sessions.end(); ++it)
   {
     CMythSession* session = *it;
     if (session->CanSupport(url))
@@ -358,15 +358,15 @@ void CMythSession::SetSeasonAndEpisode(const cmyth_proginfo_t &program, int *sea
   return;
 }
 
-CMythSession::CMythSession(const CURL& url) : CThread("MythSession")
+CMythSession::CMythSession(const CURL& url) : CThread("MythSession"),
+  m_hostname(url.GetHostName()),
+  m_username(url.GetUserName() == "" ? MYTH_DEFAULT_USERNAME : url.GetUserName()),
+  m_password(url.GetPassWord() == "" ? MYTH_DEFAULT_PASSWORD : url.GetPassWord())
 {
   m_control   = NULL;
   m_event     = NULL;
   m_database  = NULL;
   m_listener  = NULL;
-  m_hostname  = url.GetHostName();
-  m_username  = url.GetUserName() == "" ? MYTH_DEFAULT_USERNAME : url.GetUserName();
-  m_password  = url.GetPassWord() == "" ? MYTH_DEFAULT_PASSWORD : url.GetPassWord();
   m_port      = url.HasPort() ? url.GetPort() : MYTH_DEFAULT_PORT;
   m_timestamp = XbmcThreads::SystemClockMillis();
   m_dll = new DllLibCMyth;

--- a/xbmc/filesystem/NFSDirectory.cpp
+++ b/xbmc/filesystem/NFSDirectory.cpp
@@ -57,7 +57,7 @@ bool CNFSDirectory::GetDirectoryFromExportList(const std::string& strPath, CFile
   std::list<std::string> exportList=gNfsConnection.GetExportList(url);
   std::list<std::string>::iterator it;
   
-  for(it=exportList.begin();it!=exportList.end();it++)
+  for(it=exportList.begin();it!=exportList.end();++it)
   {
       std::string currentExport(*it);     
       URIUtils::RemoveSlashAtEnd(nonConstStrPath);

--- a/xbmc/filesystem/NFSFile.cpp
+++ b/xbmc/filesystem/NFSFile.cpp
@@ -137,7 +137,7 @@ void CNfsConnection::clearMembers()
 void CNfsConnection::destroyOpenContexts()
 {
   CSingleLock lock(openContextLock);
-  for(tOpenContextMap::iterator it = m_openContextMap.begin();it!=m_openContextMap.end();it++)
+  for(tOpenContextMap::iterator it = m_openContextMap.begin();it!=m_openContextMap.end();++it)
   {
     m_pLibNfs->nfs_destroy_context(it->second.pContext);
   }
@@ -258,7 +258,7 @@ bool CNfsConnection::splitUrlIntoExportAndPath(const CURL& url,std::string &expo
       
       std::list<std::string>::iterator it;
       
-      for(it=exportList.begin();it!=exportList.end();it++)
+      for(it=exportList.begin();it!=exportList.end();++it)
       {
         //if path starts with the current export path
         if(StringUtils::StartsWith(path, *it))
@@ -291,12 +291,11 @@ bool CNfsConnection::splitUrlIntoExportAndPath(const CURL& url,std::string &expo
 bool CNfsConnection::Connect(const CURL& url, std::string &relativePath)
 {
   CSingleLock lock(*this);
-  bool ret = false;
   int nfsRet = 0;
   std::string exportPath;
 
   resolveHost(url);
-  ret = splitUrlIntoExportAndPath(url, exportPath, relativePath);
+  bool ret = splitUrlIntoExportAndPath(url, exportPath, relativePath);
   
   if( (ret && (exportPath != m_exportPath  ||
        url.GetHostName() != m_hostName))    ||
@@ -374,7 +373,7 @@ void CNfsConnection::CheckIfIdle()
   {
     CSingleLock lock(keepAliveLock);
     //handle keep alive on opened files
-    for( tFileKeepAliveMap::iterator it = m_KeepAliveTimeouts.begin();it!=m_KeepAliveTimeouts.end();it++)
+    for( tFileKeepAliveMap::iterator it = m_KeepAliveTimeouts.begin();it!=m_KeepAliveTimeouts.end();++it)
     {
       if(it->second.refreshCounter > 0)
       {

--- a/xbmc/filesystem/PipeFile.cpp
+++ b/xbmc/filesystem/PipeFile.cpp
@@ -214,7 +214,7 @@ void CPipeFile::RemoveListener(IPipeListener *l)
     if ( (*i) == l)
       i = m_listeners.erase(i);
     else
-      i++;
+      ++i;
   }
 }
 

--- a/xbmc/filesystem/PipesManager.cpp
+++ b/xbmc/filesystem/PipesManager.cpp
@@ -257,7 +257,7 @@ void Pipe::RemoveListener(IPipeListener *l)
     if ( (*i) == l)
       i = m_listeners.erase(i);
     else
-      i++;
+      ++i;
   }
 }
 

--- a/xbmc/filesystem/RarFile.cpp
+++ b/xbmc/filesystem/RarFile.cpp
@@ -557,7 +557,7 @@ void CRarFile::InitFromUrl(const CURL& url)
 
   m_bFileOptions = 0;
 
-  for( vector<std::string>::iterator it = options.begin();it != options.end(); it++)
+  for( vector<std::string>::iterator it = options.begin();it != options.end(); ++it)
   {
     size_t iEqual = (*it).find('=');
     if(iEqual != std::string::npos)

--- a/xbmc/filesystem/RarManager.cpp
+++ b/xbmc/filesystem/RarManager.cpp
@@ -461,7 +461,7 @@ void CRarManager::ClearCache(bool force)
 #ifdef HAS_FILESYSTEM_RAR
   CSingleLock lock(m_CritSection);
   map<std::string, pair<ArchiveList_struct*,vector<CFileInfo> > >::iterator j;
-  for (j = m_ExFiles.begin() ; j != m_ExFiles.end() ; j++)
+  for (j = m_ExFiles.begin() ; j != m_ExFiles.end() ; ++j)
   {
 
     for (vector<CFileInfo>::iterator it2 = j->second.second.begin(); it2 != j->second.second.end(); ++it2)

--- a/xbmc/filesystem/SAPDirectory.cpp
+++ b/xbmc/filesystem/SAPDirectory.cpp
@@ -332,7 +332,7 @@ bool CSAPSessions::ParseAnnounce(char* data, int len)
   }
 
   // check if we can find this session in our cache
-  for(std::vector<CSession>::iterator it = m_sessions.begin(); it != m_sessions.end(); it++)
+  for(std::vector<CSession>::iterator it = m_sessions.begin(); it != m_sessions.end(); ++it)
   {
     if(it->origin         == header.origin
     && it->msgid          == header.msgid
@@ -496,7 +496,7 @@ namespace XFILE
       g_sapsessions.Create();
 
     // check if we can find this session in our cache
-    for(std::vector<CSAPSessions::CSession>::iterator it = g_sapsessions.m_sessions.begin(); it != g_sapsessions.m_sessions.end(); it++)
+    for(std::vector<CSAPSessions::CSession>::iterator it = g_sapsessions.m_sessions.begin(); it != g_sapsessions.m_sessions.end(); ++it)
     {
 
       if(it->payload_type != "application/sdp")

--- a/xbmc/filesystem/SAPFile.cpp
+++ b/xbmc/filesystem/SAPFile.cpp
@@ -49,7 +49,7 @@ bool CSAPFile::Open(const CURL& url)
   std::string path = url.Get();
 
   CSingleLock lock(g_sapsessions.m_section);
-  for(vector<CSAPSessions::CSession>::iterator it = g_sapsessions.m_sessions.begin(); it != g_sapsessions.m_sessions.end(); it++)
+  for(vector<CSAPSessions::CSession>::iterator it = g_sapsessions.m_sessions.begin(); it != g_sapsessions.m_sessions.end(); ++it)
   {
     if(it->path == path)
     {
@@ -70,7 +70,7 @@ bool CSAPFile::Exists(const CURL& url)
   std::string path = url.Get();
 
   CSingleLock lock(g_sapsessions.m_section);
-  for(vector<CSAPSessions::CSession>::iterator it = g_sapsessions.m_sessions.begin(); it != g_sapsessions.m_sessions.end(); it++)
+  for(vector<CSAPSessions::CSession>::iterator it = g_sapsessions.m_sessions.begin(); it != g_sapsessions.m_sessions.end(); ++it)
   {
     if(it->path == path)
       return true;
@@ -94,7 +94,7 @@ int CSAPFile::Stat(const CURL& url, struct __stat64* buffer)
 
 
   CSingleLock lock(g_sapsessions.m_section);
-  for(vector<CSAPSessions::CSession>::iterator it = g_sapsessions.m_sessions.begin(); it != g_sapsessions.m_sessions.end(); it++)
+  for(vector<CSAPSessions::CSession>::iterator it = g_sapsessions.m_sessions.begin(); it != g_sapsessions.m_sessions.end(); ++it)
   {
     if(it->path == path)
     {
@@ -164,7 +164,7 @@ bool CSAPFile::Delete(const CURL& url)
   std::string path = url.Get();
 
   CSingleLock lock(g_sapsessions.m_section);
-  for(vector<CSAPSessions::CSession>::iterator it = g_sapsessions.m_sessions.begin(); it != g_sapsessions.m_sessions.end(); it++)
+  for(vector<CSAPSessions::CSession>::iterator it = g_sapsessions.m_sessions.begin(); it != g_sapsessions.m_sessions.end(); ++it)
   {
     if(it->path == path)
     {

--- a/xbmc/filesystem/SFTPFile.cpp
+++ b/xbmc/filesystem/SFTPFile.cpp
@@ -48,7 +48,7 @@ using namespace XFILE;
 using namespace std;
 
 
-static std::string CorrectPath(const std::string path)
+static std::string CorrectPath(const std::string& path)
 {
   if (path == "~")
     return "./";
@@ -245,17 +245,15 @@ bool CSFTPSession::GetDirectory(const std::string &base, const std::string &fold
 
 bool CSFTPSession::DirectoryExists(const char *path)
 {
-  bool exists = false;
   uint32_t permissions = 0;
-  exists = GetItemPermissions(path, permissions);
+  bool exists = GetItemPermissions(path, permissions);
   return exists && S_ISDIR(permissions);
 }
 
 bool CSFTPSession::FileExists(const char *path)
 {
-  bool exists = false;
   uint32_t permissions = 0;
-  exists = GetItemPermissions(path, permissions);
+  bool exists = GetItemPermissions(path, permissions);
   return exists && S_ISREG(permissions);
 }
 
@@ -566,7 +564,7 @@ void CSFTPSessionManager::ClearOutIdleSessions()
     if (iter->second->IsIdle())
       sessions.erase(iter++);
     else
-      iter++;
+      ++iter;
   }
 }
 

--- a/xbmc/filesystem/SMBDirectory.cpp
+++ b/xbmc/filesystem/SMBDirectory.cpp
@@ -279,7 +279,6 @@ int CSMBDirectory::OpenDir(const CURL& url, std::string& strAuth)
 
 bool CSMBDirectory::Create(const CURL& url2)
 {
-  bool success = true;
   CSingleLock lock(smb);
   smb.Init();
 
@@ -288,7 +287,7 @@ bool CSMBDirectory::Create(const CURL& url2)
   std::string strFileName = smb.URLEncode(url);
 
   int result = smbc_mkdir(strFileName.c_str(), 0);
-  success = (result == 0 || EEXIST == errno);
+  bool success = (result == 0 || EEXIST == errno);
   if(!success)
     CLog::Log(LOGERROR, "%s - Error( %s )", __FUNCTION__, strerror(errno));
 

--- a/xbmc/filesystem/SMBFile.cpp
+++ b/xbmc/filesystem/SMBFile.cpp
@@ -208,7 +208,7 @@ std::string CSMB::URLEncode(const CURL &url)
   std::vector<std::string> parts;
   std::vector<std::string>::iterator it;
   StringUtils::Tokenize(url.GetFileName(), parts, "/");
-  for( it = parts.begin(); it != parts.end(); it++ )
+  for( it = parts.begin(); it != parts.end(); ++it )
   {
     flat += "/";
     flat += URLEncode((*it));

--- a/xbmc/filesystem/SmartPlaylistDirectory.cpp
+++ b/xbmc/filesystem/SmartPlaylistDirectory.cpp
@@ -82,7 +82,7 @@ namespace XFILE
 
     // get all virtual folders and add them to the item list
     playlist.GetVirtualFolders(virtualFolders);
-    for (std::vector<std::string>::const_iterator virtualFolder = virtualFolders.begin(); virtualFolder != virtualFolders.end(); virtualFolder++)
+    for (std::vector<std::string>::const_iterator virtualFolder = virtualFolders.begin(); virtualFolder != virtualFolders.end(); ++virtualFolder)
     {
       CFileItemPtr pItem = CFileItemPtr(new CFileItem(*virtualFolder, true));
       IFileDirectory *dir = CFileDirectoryFactory::Create(pItem->GetURL(), pItem.get());

--- a/xbmc/filesystem/StackDirectory.cpp
+++ b/xbmc/filesystem/StackDirectory.cpp
@@ -70,7 +70,7 @@ namespace XFILE
         RegExps.push_back(tempRE);
       else
         CLog::Log(LOGERROR, "Invalid video stack RE (%s). Must have exactly 4 captures.", itRegExp->c_str());
-      itRegExp++;
+      ++itRegExp;
     }
     return GetStackedTitlePath(strPath, RegExps);
   }
@@ -147,7 +147,7 @@ namespace XFILE
           }
         }
         offset = 0;
-        itRegExp++;
+        ++itRegExp;
       }
       if (!strCommonDir.empty() && !strStackTitle.empty())
         strStackTitlePath = strCommonDir + strStackTitle;
@@ -188,7 +188,7 @@ namespace XFILE
       return false;
 
     // because " , " is used as a seperator any "," in the real paths are double escaped
-    for (vector<std::string>::iterator itPath = vecPaths.begin(); itPath != vecPaths.end(); itPath++)
+    for (vector<std::string>::iterator itPath = vecPaths.begin(); itPath != vecPaths.end(); ++itPath)
       StringUtils::Replace(*itPath, ",,", ",");
 
     return true;

--- a/xbmc/filesystem/TuxBoxDirectory.cpp
+++ b/xbmc/filesystem/TuxBoxDirectory.cpp
@@ -194,7 +194,7 @@ void CTuxBoxDirectory::GetRootAndChildStringEnigma2(std::string& strBQRequest, s
   strXMLChildString = StringUtils::Format("e2bouquet");
 }
 
-bool CTuxBoxDirectory::GetRootAndChildString(const std::string strPath, std::string& strBQRequest, std::string& strXMLRootString, std::string& strXMLChildString )
+bool CTuxBoxDirectory::GetRootAndChildString(const std::string& strPath, std::string& strBQRequest, std::string& strXMLRootString, std::string& strXMLChildString )
 {
   //Advanced Settings: RootMode! Movies:
   if(g_advancedSettings.m_iTuxBoxDefaultRootMenu == 3) //Movies! Fixed-> mode=3&submode=4

--- a/xbmc/filesystem/TuxBoxDirectory.h
+++ b/xbmc/filesystem/TuxBoxDirectory.h
@@ -36,7 +36,7 @@ namespace XFILE
       virtual bool AllowAll() const { return true; }
       virtual DIR_CACHE_TYPE GetCacheType(const CURL& url) const { return DIR_CACHE_ALWAYS; };
     private:
-      bool GetRootAndChildString(const std::string strPath, std::string& strBQRequest, std::string& strXMLRootString, std::string& strXMLChildString );
+      bool GetRootAndChildString(const std::string& strPath, std::string& strBQRequest, std::string& strXMLRootString, std::string& strXMLChildString );
       void GetRootAndChildStringEnigma2(std::string& strBQRequest, std::string& strXMLRootString, std::string& strXMLChildString );
   };
 }

--- a/xbmc/filesystem/UPnPDirectory.cpp
+++ b/xbmc/filesystem/UPnPDirectory.cpp
@@ -343,7 +343,7 @@ CUPnPDirectory::GetDirectory(const CURL& url, CFileItemList &items)
 
         NPT_String max_string = "";
         int        max_count  = 0;
-        for(std::map<NPT_String, int>::iterator it = classes.begin(); it != classes.end(); it++)
+        for(std::map<NPT_String, int>::iterator it = classes.begin(); it != classes.end(); ++it)
         {
           if(it->second > max_count)
           {

--- a/xbmc/filesystem/VTPDirectory.cpp
+++ b/xbmc/filesystem/VTPDirectory.cpp
@@ -46,7 +46,7 @@ bool CVTPDirectory::GetChannels(const std::string& base, CFileItemList &items)
     return false;
 
   vector<CVTPSession::Channel>::iterator it;
-  for(it = channels.begin(); it != channels.end(); it++)
+  for(it = channels.begin(); it != channels.end(); ++it)
   {
     CFileItemPtr item(new CFileItem("", false));
 

--- a/xbmc/filesystem/VTPSession.cpp
+++ b/xbmc/filesystem/VTPSession.cpp
@@ -367,7 +367,7 @@ bool CVTPSession::GetChannels(std::vector<Channel> &channels)
   if(!SendCommand("LSTC", code, lines))
     return false;
 
-  for(std::vector<std::string>::iterator it = lines.begin(); it != lines.end(); it++)
+  for(std::vector<std::string>::iterator it = lines.begin(); it != lines.end(); ++it)
   {
     std::string& data(*it);
     size_t space, colon;

